internal/cli/daemon/client.go

```diff
@@ -0,0 +1,97 @@
+package daemon
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"octavius/internal/cli/client"
+	"octavius/internal/cli/config"
+	"octavius/pkg/protobuf"
+	"time"
+
+	"github.com/gogo/protobuf/jsonpb"
+	"google.golang.org/grpc"
+)
+
+type Client interface {
+	StartClient() error
+	CreateMetadata(metadataFile string) error
+}
+
+type octaviusClient struct {
+	octaviusConfigLoader  config.Loader
+	grpcClient            client.Client
+	CPHost                string
+	emailId               string
+	accessToken           string
+	connectionTimeoutSecs time.Duration
+}
+
+func NewClient(clientConfigLoader config.Loader) Client {
+	return &octaviusClient{
+		octaviusConfigLoader: clientConfigLoader,
+	}
+}
+
+func (c *octaviusClient) StartClient() error {
+	err := c.loadOctaveConfig()
+	if err != nil {
+		return err
+	}
+
+	conn, err := grpc.Dial(c.CPHost, grpc.WithInsecure())
+	if err != nil {
+		return err
+	}
+	grpcClient := protobuf.NewOctaviusServicesClient(conn)
+	client := client.NewGrpcClient(grpcClient)
+	c.grpcClient = client
+	return nil
+}
+
+func (c *octaviusClient) loadOctaveConfig() error {
```

Comment:
since `loadOctaveConfig` a local function, should we add `c` as a parameter and not a method object?
Also proper naming like `loadOctaviusConfig` should be used

Reply:
As we are making changes inside the object it makes more sense to make it a method of the class rather than pass an argument. Naming sounded cool, will change it.